source/tutorials/learning-journey/shell-dot-nix.md

```diff
@@ -0,0 +1,159 @@
+
+Setting an environment variable in via `shell.nix` is trivial.
+Any attribute in the `mkShell` function call that `mkShell` doesn't recognize as a reserved attribute name will be set to an environment variable in the shell environment.
+The attributes that are reserved are listed in the [Nixpkgs manual][mkshell_attrs] and include `packages`, `name`, and several others.
+
+[mkshell_attrs]: https://nixos.org/manual/nixpkgs/stable/#sec-pkgs-mkShell-attributes
+
+Let's say you wanted to set the database user (`DB_USER`) and password (`DB_PASSWORD`) via environment variables in your `shell.nix` file.
+This is how that would look:
+```nix
+let
+    pkgs = import <nixpkgs> {};
+in
+    pkgs.mkShell {
+        packages = [
+          pkgs.python310
+          pkgs.curl
+        ];
+
+        # Database username and password
+        DB_USER = "db_user";
+        DB_PASSWORD = "super secret don't look";
+    }
+```
+
+Not only can you set new environment variables, but you can overwrite _existing_ environment variables.
+For instance, the shell prompt format is set by the `PS1` environment variable.
+In order to set your own prompt you can simply set the `PS1` attribute in the attribute set passed to the `mkShell` command
+To set the shell prompt to the format `<username>@<hostname> [myEnv] $ ` the `shell.nix` file would look like this:
+
+FIXME: This doesn't actually set the prompt for some reason
+
+```nix
+let
+    pkgs = import <nixpkgs> {};
+in
+    pkgs.mkShell {
+        packages = [
+          pkgs.python310
+          pkgs.curl
+        ];
+
+        # Database username and password
+        DB_USER = "db_user";
+        DB_PASSWORD = "super secret don't look";
+
+        # Set the shell prompt to '<username>@<hostname> [myEnv] $ '
+        PS1 = "\u@\h [myEnv] $ ";
+    }
+```
+
+
+## Startup commands
+You may want to perform some initialization before entering the shell environment (for example, maybe you want to ensure that a file exists).
+Commands you'd like to run before entering the shell environment can be placed in the `shellHook` attribute of the attribute set provided to the `mkShell` function.
+To ensure that a file `should_exist.txt` exists, the `shell.nix` file would look like this:
+```nix
+let
+    pkgs = import <nixpkgs> {};
+in
+    pkgs.mkShell {
+        packages = [
+          pkgs.python310
+          pkgs.curl
+        ];
+
+        # Database username and password
+        DB_USER = "db_user";
+        DB_PASSWORD = "super secret don't look";
+
+        # Ensure that 'should_exist.txt' exists
+        shellHook = ''
+        touch should_exist.txt
+        '';
+    }
+```
+
+
+## Where to next?
+
+<!-- Is there something the reader should read next? -->
+<!-- Are there other topics they should explore next? -->
+<!-- Provide links to other resources that might be relevant. -->
+WIP
```

Comment:
Let's not include too many things at once, merge it once ready without more things.

Can include `inputsFrom` in a XML comment here for the future.